### ISSUE TITLE
Question about the results : MoFlow and IMLE

### ISSUE THREAD

**Issue by gaotianci99** [felix-yuxiang/MoFlow#9]
Thank you for your excellent work!

Does “teacher sampling results” refer to the MoFlow metrics reported in the tables, and does “student sampling results” correspond to the IMLE metrics?

When evaluating on the ETH subset using the following command (Teacher Sampling Stage):
```bash
python3 eval_eth.py \
  --ckpt_path /home/zhengda/GTC/MoFlow-main/results_eth_ucy/cor_fm/teaExp01_eth/models/checkpoint_best.pt \
  --subset eth \
  --rotate \
  --rotate_time_frame 6 \
  --batch_size 1000 \
  --sampling_steps 100 \
  --solver lin_poly \
  --lin_poly_p 5 \
  --lin_poly_long_step 1000 \
  --save_samples \
  --eval_on_train
```
I noticed that the test results differ significantly from those reported in Table 2 of the paper. Could you help clarify why this discrepancy might occur?
<img width="1343" height="757" alt="Image" src="https://github.com/user-attachments/assets/dabfc138-beac-4095-81b5-5ab316377e23" />

**Comment by felix-yuxiang** [comment]
Hi @gaotianci99 , I’m glad to hear that you’re interested in our work. The discrepancy you’re seeing is due to the fact that you are currently evaluating on the training set, which means the reported error is the training error.

It may help to recall why our pipeline includes the _teacher sampling phase_. IMLE requires drawing K samples from the teacher MoFlow model (see Algorithm 1, line 6 in our paper). The provided script is therefore intended to save these teacher samples for the training set.

To properly evaluate MoFlow on the ETH dataset, you should disable `--eval_on_train`. This will allow you to run evaluation on the test set instead. You can also shut down `--save_samples` if you prefer.

I highly recommend reading this suggestion before running the scripts. This will save tons of time. Cheers. https://github.com/DSL-Lab/MoFlow?tab=readme-ov-file#suggestions
